tools/setup_helpers/generate_code.py

```diff
@@ -172,16 +181,24 @@ def main() -> None:
     )
     options = parser.parse_args()
 
-    generate_code(
-        options.ninja_global,
-        options.native_functions_path,
-        options.install_dir,
-        options.subset,
-        options.disable_autograd,
-        options.force_schema_registration,
-        # options.selected_op_list
-        operator_selector=get_selector(options.selected_op_list_path, options.operators_yaml_path),
-    )
+    with tempfile.TemporaryDirectory() as autograd_dir_str:
```

Comment:
Could you give some details on why we need to move all these files around?
Also where are these things used?

Reply:
So the only reason I moved them around was to avoid modifying more of the code downstream, i.e. so downstream can just treat this like a regular directory unconditionally.

I could instead modify the bowels to read from pkg_resources instead of moving them to a temporary directory. What do you think?

If you would rather me keep it this way, I'm happy to leave a comment explaining why we're doing this.

I do want to avoid having multiple ways of accessing these templates. I'd like to have a consistent way of getting them for cmake, buck, and bazel.

Reply:
I agree that moving them around to avoid having weird code to access them is good.
A slightly more detailed comment here would be great for the next person that will have to touch this code!

Reply:
Oops. I did have a more detailed comment elsewhere but it got orphaned when I moved the code. How does this look now?

Reply:
I don't see any extra comment in this PR?

Reply:
https://github.com/pytorch/pytorch/pull/74672/files#diff-a4e30a25170f1f3331c892756b11ac80d9ebc30ba7ca4d1cce4d757cde675472R49 up in generate_code.